ui/src/components/AddServerDialog.tsx

```diff
@@ -309,6 +309,11 @@ export function AddServerDialog({ open, onOpenChange, onAddServer, onError }: Ad
         url: url.trim(),
       };
 
+      // Add http:// protocol if missing
+      if (params.url && !params.url.match(/^[a-z]+:\/\//i)) {
+        params.url = `http://${params.url}`;
+      }
+
```

Comment:
@peterj what do you think about this. It seems to me like we should potentially just validate rather than magically prepend.

Reply:
I agree. I think we should just error out instead of prepending (e.g. call `setError("Please enter a valid URL")` or something likethat)